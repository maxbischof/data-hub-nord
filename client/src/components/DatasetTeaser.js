import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

export default function DatasetTeaser({ imagePath, title, description }) {
  return (
    <StyledDatasetTeaser>
      <Title>{title}</Title>
      <hr />
      <Description>
        {description.length > 70
          ? description.substring(0, 67) + '...'
          : description}
      </Description>
    </StyledDatasetTeaser>
  )
}

DatasetTeaser.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
}

const StyledDatasetTeaser = styled.article`
  margin: 25px;
  max-width: 300px;
`

const Title = styled.h3`
  font-size: 20px;
  font-weight: 700;
  margin: 10px 0 0 0;
`

const Description = styled.p`
  font-size: 17px;
  font-weight: 400;
  margin: 10px 0 0 0;
  word-wrap: break-word;
`
