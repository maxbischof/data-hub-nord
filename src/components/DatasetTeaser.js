import React from 'react'
import DatasetImage from './DatasetImage'
import styled from 'styled-components'

export default function DatasetTeaser ({ imagePath, title, description }) {

  return (
    <StyledDatasetTeaser>
      <DatasetImage path={imagePath} />
      <Title>{title}</Title>
      <Description>{description.length > 70 ? description.substring(0, 67) + "..." : description}</Description>
    </StyledDatasetTeaser>
  )
}

const StyledDatasetTeaser = styled.article`
  margin: 0 0 50px 0;
`

const Title = styled.h3`
  font-size: 20px;
  font-weight: 700;
  margin: 10px 0 0 0;
`

const Description = styled.p`
  font-size: 17px;
  font-weight: 400;
  margin: 10px 0 0 0;
  word-wrap: break-word;
`