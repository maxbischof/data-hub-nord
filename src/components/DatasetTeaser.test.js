import React from "react"
import { render } from "@testing-library/react"
import renderer from 'react-test-renderer'
import DatasetTeaser from "./DatasetTeaser"

it("has title and description < 70 characters", () => {
  const { getByText } = render(
    <DatasetTeaser
      imagePath="/Users/student/neuefische/data-hub-nord/src/images/sea.jpg"
      title="Seehundbänke an der Nordsee"
      description="Lage der Seehundbänke in der westlichen Deutschen Bucht erfasst durch mehrere Forschungsschiffe"
    />
  )

  const title = getByText("Seehundbänke an der Nordsee")
  expect(title).toBeInTheDocument()

  const description = getByText("Lage der Seehundbänke in der westlichen Deutschen Bucht erfasst dur...")
  expect(description).toBeInTheDocument()
})

it('renders correctly', () => {
  const tree = renderer.create(
    <DatasetTeaser
      imagePath="/Users/student/neuefische/data-hub-nord/src/images/sea.jpg"
      title="Seehundbänke an der Nordsee"
      description="Lage der Seehundbänke in der westlichen Deutschen Bucht erfasst durch mehrere Forschungsschiffe"
    />
  )
  expect(tree).toMatchSnapshot()
})