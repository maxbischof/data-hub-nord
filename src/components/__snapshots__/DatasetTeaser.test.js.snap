// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="sc-AxirZ bMOfdg"
>
  <img
    className="sc-AxjAm dqGSZK"
    src="/Users/student/neuefische/data-hub-nord/src/images/sea.jpg"
  />
  <h3
    className="sc-AxiKw kaYFBU"
  >
    Seehundbänke an der Nordsee
  </h3>
  <p
    className="sc-AxhCb gIDQGb"
  >
    Lage der Seehundbänke in der westlichen Deutschen Bucht erfasst dur...
  </p>
</article>
`;
